vulture/core.py

```diff
@@ -191,11 +200,11 @@ def get_list(typ):
         self.ignore_names = ignore_names or []
         self.ignore_decorators = ignore_decorators or []
 
-        self.filename = ""
+        self.filename = pathlib.Path()
         self.code = []
         self.found_dead_code_or_error = False
 
-    def scan(self, code, filename=""):
+    def scan(self, code, filename=DEFAULT_FILE_PATH):
```

Comment:
filename can stay a string, because it's only used for printing and never for accessing files, I think.

Reply:
It would be used to find the relative path in `utils.format_path()` though..

Reply:
Ah, good point! Then let's convert accept strings and Path objects for scan() as well and convert the argument to a Path object at the beginning of the method. The default `filename=""` (empty string) can remain unchanged, no need for DEFAULT_FILE_PATH.

Reply:
But with the changes in this PR, the `filename` arg only accepts `Path` objects. Should we still allow `str`s? Or maybe this got to be added for leaving provision for changes allowing `str` in the future?

Also, we could remove the `DEFAULT_FILE_PATH` and use `None` instead.

Reply:
Yes, let's allow strings and Path objects. And remove DEFAULT_FILE_PATH and use `filename=""` as before.